ethereum/test/bridge.js

```diff
@@ -614,6 +615,147 @@ contract("Bridge", function () {
         assert.equal(bridgeBalanceAfter.toString(10), "0");
     })
 
+    it("should deposit and log transfer with payload correctly", async function () {
+        const accounts = await web3.eth.getAccounts();
+        const amount = "1000000000000000000";
+        const fee = "100000000000000000";
+
+        // mint and approve tokens
+        const token = new web3.eth.Contract(TokenImplementation.abi, TokenImplementation.address);
+        await token.methods.mint(accounts[0], amount).send({
+            value: 0,
+            from: accounts[0],
+            gasLimit: 2000000
+        });
+        await token.methods.approve(TokenBridge.address, amount).send({
+            value: 0,
+            from: accounts[0],
+            gasLimit: 2000000
+        });
+
+        // deposit tokens
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+
+        const accountBalanceBefore = await token.methods.balanceOf(accounts[0]).call();
+        const bridgeBalanceBefore = await token.methods.balanceOf(TokenBridge.address).call();
+
+        assert.equal(bridgeBalanceBefore.toString(10), "0");
```

Comment:
not really, since it would depend on the results of the previous tests and therefore incorrectly causes failures based on the state of the account. the real test is that it differs by the amount after the transfer